Analysis/src/EventTags.cpp

```diff
@@ -94,8 +141,10 @@ EventTags EventTagCreator::CreateEventTags(const DataId& dataId_base, float weig
         if(!category.Contains(static_cast<size_t>(num_central_jets), bjet_counts, is_vbf, is_boosted, vbf_cat.second))
             continue;
         EventSubCategory evtSubCategory;
-        const float btag_sf = category.HasBtagConstraint() && !is_data ? get_weight_btag(category.BtagWP()) : 1.f;
-        const float cat_weight = weight * btag_sf;
+        const float btag_sf=1.f;
+
+        const float btag_sf = category.HasBtagConstraint() && !is_data ? get_weight_btag(category.BtagWP(), dataId_base.Get<UncertaintySource>(), dataId_base.Get<UncertaintyScale>()) : 1.f;
+        const float cat_weight = btag_uncs.count(dataId_base.Get<UncertaintySource>())? weight * btag_sf * unc_map.at(ategory.BtagWP(), dataId_base.Get<UncertaintySource>(), dataId_base.Get<UncertaintyScale>()) : weight * btag_sf ;
```

Comment:
this looks too complicated, you alreay applied conditions when computing btag_sf, why it needs to be repeated here?